cmd/gpu_fakedev/gpu_fakedev.go

```diff
@@ -148,6 +149,46 @@ func addSysfsDriTree(root string, opts *genOptions, i int) error {
 	return nil
 }
 
+func addBusFiles(root string, opts *genOptions, i int) error {
+	pciName := fmt.Sprintf("0000:00:0%d.0", i)
+	base := filepath.Join(root, "bus/pci/drivers/i915", pciName)
+
+	if err := os.MkdirAll(base, dirMode); err != nil {
+		return err
+	}
+	opts.dirs++
+
+	data := []byte("0x4905")
+	file := filepath.Join(base, "device")
+
+	if err := os.WriteFile(file, data, fileMode); err != nil {
+		return err
+	}
+	opts.files++
+
+	drm := filepath.Join(root, "bus/pci/drivers/i915", pciName, "drm")
+	if err := os.MkdirAll(drm, dirMode); err != nil {
+		return err
+	}
+	opts.dirs++
+
+	cardName := fmt.Sprintf("card%d", i)
+	file = filepath.Join(drm, cardName)
+	if err := os.WriteFile(file, data, fileMode); err != nil {
+		return err
+	}
+	opts.files++
+
+	renderDName := fmt.Sprintf("renderD%d", i+128)
+	file = filepath.Join(drm, renderDName)
+	if err := os.WriteFile(file, data, fileMode); err != nil {
+		return err
+	}
+	opts.files++
```

Comment:
These are dirs in sysfs, not files.

IMHO best would be to to split card/renderD subdir creation from `addSysfsDriTree()` to its own function, e.g.
```
// addCardAndRender creates card<x> & renderD<xxx> subdirs to given sysfs parent directory
addCardAndRender(parentdir string, cardidx, renderidx uint, opts *genOptions)
```
and call it from both `addSysfsDriTree()` and `addBusFiles()`.

After that, it may also make sense rename latter to `addSysfsBusTree()`.

Reply:
This part what I wrote, is mostly broken, but it passes the DRA parsing. That's all it achieves really. I'm not sure if what you suggest will be sufficient to fix things so that the fake files would appear with DRA, probably not. But I suppose the suggestions you make are proper and thus worth doing.